code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -155,14 +155,10 @@
 	visible_message("<span class='xenowarning'>\The [src] spews forth a wide cone of acid!</span>", \
 	"<span class='xenowarning'>You spew forth a cone of acid!</span>", null, 5)
 
-	speed += 2
+	add_slowdown(2) //Getting rid of the shit code that allowed Praes to go sonic if they regained temp lost speed during upgrade
```

Comment:
relying on life to fix slowdown is terrible.
assuming theres no complications with pheromones giving speed increases then you should use the caste speed as the base value to reset from.
A lot of these slowdown handling procs dont consider xeno datum stats

Reply:
Should be fine now